v2/samples/dataprotection/v1api/v1api20230101_backupvaultsbackupinstance.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: dataprotection.azure.com/v1api20230101
+kind: BackupVaultsBackupInstance
+metadata:
+  name: asotestbackupinstance
+  namespace: default
+spec:
+  owner:
+    name: asotestbackupvault
+  tags:
+    tag1: value1
+  properties:
+    friendlyName: asotestbackupinstance
+    dataSourceInfo:
+      objectType: Datasource
+      resourceID: "/subscriptions/f0c630e0-2995-4853-b056-0b3c09cb673f/resourceGroups/t-agrawals/providers/Microsoft.ContainerService/managedClusters/testAKScluster"
```

Comment:
There is no resourceID field anymore, it's `resourceReference`, so you should use that instead here.

Also: If the user is already providing an ID, which contains: `resourceName`, resourceType, location, etc all embedded in it, why do they also need to fill out all these other fields? Can't they all be inferred from the resourceReference (== resourceID in ARM) field?

Reply:
This has been rectified in new sample yaml files. 

So, ResourceID is a required ALWAYS field, and the rest of the other fields, including ResourceLocation, ResourceName, and ResourceType, are required default. 

The need for providing fields other than ResourceID isn't mandatory, but good to have. Did you mean extracting values of other fields from URl? If so, in my humble opinion, this ain't possible in yaml. However, would like to know more insights if required.